src/polycubed/src/utils/veth_peer.h

```diff
@@ -0,0 +1,70 @@
+#include <stdlib.h>
+#include <vector>
+#include <sstream>
+#include <iostream>
+#include <stdexcept>
+#include <thread>
+#include <sys/ioctl.h>
+#include <linux/limits.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <net/if.h>
+#include <arpa/inet.h>
+#include <netinet/ether.h>
+#include <netlink/socket.h>
+#include <netlink/netlink.h>
+#include <netlink/route/link.h>
+#include <netlink/route/link/veth.h>
+
+#define NLMSG_TAIL(nmsg) \
+        ((struct rtattr *) (((char *) (nmsg)) + NLMSG_ALIGN((nmsg)->nlmsg_len)))
+
+struct nl_req {
+	struct nlmsghdr nlmsg;
+	struct ifinfomsg ifinfomsg;
+};
+
+struct nl_ipreq {
+	struct nlmsghdr nlmsg;
+	struct ifaddrmsg ifaddrmsg;
+};
+
+class VethPeer {
+  public:
+    VethPeer();
+    ~VethPeer();
+    void resetPeer();
+
+    void setName(const std::string &name);
+    void setStatus(const bool status);
+    void setNamespace(const std::string &ns);
+    void setMAC(const std::string &mac);
+    void setIP(const std::string &IPaddr);
+
+    std::string getName();
+    bool getStatus();
+    std::string getNamespace();
+    std::string getMAC();
+    std::string getIP();
+
+  private:
+    std::string name_;
+    bool status_;
+    std::string ns_;
+    std::string mac_;
+    std::string ip_;
+    int prefix_;
```

Comment:
is this really a good idea to save all the info in the object?, I think only name could be saved here and other parameters could be queried to the kernel.

This because the interface could be modified from the external to this would not reflect that change.